### PR TITLE
Ion Identity Molecular Network Visualization

### PR DESCRIPTION
**Pull request by robinschmid**
First visualization of ion identity molecular networks in MZmine.
It takes any RowsRelationships (Row2RowMaps) and displays them as graphs. 

### Features
- Export to graphml (open format)
- export to SVG (still needs some fixing with the layout)
- Dynamically change the color, size, or label of nodes
- Collapse ion identity networks

image

### CODE REVIEW

**Comment by Annexhc** [comment]
That looks great!

File: src/main/java/io/github/mzmine/datamodel/features/FeatureListRow.java [hunk 1]
```diff
@@ -368,6 +370,16 @@ default boolean hasMs2Fragmentation() {
...
 
+  /**
+   * The intensity summed over all features
+   * @return sum of all feature heights
+   */
+  default double getSumIntensity() {
+    return this.getFeatures().stream().filter(Objects::nonNull)
+        .filter(f -> f.getFeatureStatus() != FeatureStatus.UNKNOWN).mapToDouble(
+        Feature::getHeight).sum();
+  }
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/datamodel/features/FeatureListRow.java:381]: When is it ok to use streams and when is a for loop better? @tomas-pluskal @robinschmid 
This looks like a method that might be called often during visualisation, so maybe a raw loop is more performant?

File: src/main/java/io/github/mzmine/modules/visualization/networking/AnnotationNetworkModule.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+import io.github.mzmine.modules.MZmineRunnableModule;
+import io.github.mzmine.modules.visualization.networking.visual.FeatureNetworkTab;
+import io.github.mzmine.parameters.ParameterSet;
+import io.github.mzmine.taskcontrol.Task;
+import io.github.mzmine.util.ExitCode;
+import java.util.Collection;
+import javax.annotation.Nonnull;
+
+/**
+ * TIC/XIC visualizer using JFreeChart library
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/modules/visualization/networking/AnnotationNetworkModule.java:33]: this doesnt seem to belong here :D

File: src/main/java/io/github/mzmine/modules/visualization/networking/visual/FeatureNetworkTab.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    contentPane.setShowMs2SimEdges(ms2SimEdges);
+    contentPane.setUseMs1FeatureShapeEdges(ms1FeatureShapeEdges);
+    // last, as it recreates the graph
+    contentPane.setFeatureList(featureList);
+  }
+
+  @Nonnull
+  @Override
+  public Collection<? extends RawDataFile> getRawDataFiles() {
+    return List.of();
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/modules/visualization/networking/visual/FeatureNetworkTab.java:104]: is this equivalent to Collections.emptyList() or does it generate a new list every time?

File: src/main/java/io/github/mzmine/modules/visualization/networking/AnnotationNetworkModule.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+import io.github.mzmine.datamodel.features.ModularFeatureList;
+import io.github.mzmine.main.MZmineCore;
+import io.github.mzmine.modules.MZmineModuleCategory;
+import io.github.mzmine.modules.MZmineRunnableModule;
+import io.github.mzmine.modules.visualization.networking.visual.FeatureNetworkTab;
+import io.github.mzmine.parameters.ParameterSet;
+import io.github.mzmine.taskcontrol.Task;
+import io.github.mzmine.util.ExitCode;
+import java.util.Collection;
+import javax.annotation.Nonnull;
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/modules/visualization/networking/AnnotationNetworkModule.java:30]: wrong annotation

File: src/main/java/io/github/mzmine/modules/visualization/networking/visual/FeatureNetworkTab.java [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+import io.github.mzmine.gui.mainwindow.MZmineTab;
+import io.github.mzmine.main.MZmineCore;
+import java.util.Collection;
+import java.util.List;
+import javafx.scene.control.CheckMenuItem;
+import javafx.scene.control.Menu;
+import javafx.scene.control.MenuBar;
+import javafx.scene.layout.BorderPane;
+import javafx.scene.layout.VBox;
+import javax.annotation.Nonnull;
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/modules/visualization/networking/visual/FeatureNetworkTab.java:32]: wrong annotation

**Pull request merged**